test/parser/Makefile

```diff
@@ -0,0 +1,34 @@
+ifeq ($(origin TOP), undefined)
+	TOP = $(shell git rev-parse --show-toplevel)
+endif
+
+include $(TOP)/include.mk
+
+DIFF = diff -u
+%.golden: DIFF = true
+
+FAILED = ( mv $@ $@.fail && false )
+
+TESTS = $(wildcard *.kore)
+OUTS += $(foreach TEST, $(TESTS), $(TEST).out)
+GOLDEN += $(foreach TEST, $(TESTS), $(TEST).golden)
+
+%.kore.out : %.kore $(KORE_PARSER)
+	$(KORE_PARSER) --no-print-definition $< 2>$@ || true
+	$(DIFF) $*.kore.golden $@ || $(FAILED)
+
+%.golden: %.out
```

Comment:
why are producing golden files like this?
I thought that golden files are written mostly by hand according to expected behavior rather than what the code is producing right now (unless I understood this recipe wrong).